src/VisualStudio/Core/Def/Snippets/AbstractSnippetExpansionClientFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Snippets;
+
+internal abstract class AbstractSnippetExpansionClientFactory : ISnippetExpansionClientFactory
+{
+    protected abstract AbstractSnippetExpansionClient CreateSnippetExpansionClient(ITextView textView, ITextBuffer subjectBuffer);
+
+    public AbstractSnippetExpansionClient GetSnippetExpansionClient(ITextView textView, ITextBuffer subjectBuffer)
+    {
+        if (!textView.Properties.TryGetProperty(typeof(AbstractSnippetExpansionClient), out AbstractSnippetExpansionClient expansionClient))
```

Comment:
would it be possible to add some threading asserts here?  it's not a big deal, it just makes me feel less oogy about doing weird stuff with proeprty bags on random threads.

Reply:
➡️ Added assertion